### PR TITLE
E2E Test for Observation

### PR DESCRIPTION
**Pull request by fwpushan**
(#531 )

1. Added e2e test for observation
2. Fix some small UI issues.

### CODE REVIEW

File: app/lucy/src/app/components/base-form/location-input/location-input/location-input.component.ts [hunk 1]
```diff
@@ -221,6 +221,7 @@ export class LocationInputComponent implements OnInit {
     if (this.showX && this.showY) {
       this.areaCalculator = this.calculatePlot;
       this.headerXDimension = AreaFieldTitle.WIDTH;
+      this.headerYDimension = AreaFieldTitle.LENGTH;
```
Comment (AmirShayegh) [inline app/lucy/src/app/components/base-form/location-input/location-input/location-input.component.ts:224]: nice catch

File: app/lucy/cypress/integration/inv-add-entry/nav_plant_obs.spec.ts [hunk 2]
```diff
@@ -13,53 +16,142 @@ describe('/add', () => {
     });
     cy.viewport('macbook-15');
 
+    const testValues: any = {
```
Comment (AmirShayegh) [inline app/lucy/cypress/integration/inv-add-entry/nav_plant_obs.spec.ts:19]: I like this, looks clean

File: app/lucy/cypress/integration/inv-add-entry/nav_plant_obs.spec.ts [hunk 3]
```diff
@@ -13,53 +16,142 @@ describe('/add', () => {
...
+
+    // Range Unit Number
+    cy.get('#rangeUnitNumber').find('[placeholder="Range Unit Number"]').type('11X1{enter}', { force: true });
+
+    // Sample Identifier
+    cy.get('#sampleIdentifier').find('[placeholder="Sample Identifier"]').type('1NCC345{enter}', { force: true });
+
+
+    // Reviewing
+    cy.contains('button', 'Submit Observation ').click({ force: true });
```
Comment (AmirShayegh) [inline app/lucy/cypress/integration/inv-add-entry/nav_plant_obs.spec.ts:111]: there is a .then({ })
Do you think it would apply to this case?

**Review by AmirShayegh — APPROVED** [review]

File: app/lucy/cypress/integration/inv-add-entry/nav_plant_obs.spec.ts (hunk 3 above)
Comment (fwpushan) [reply app/lucy/cypress/integration/inv-add-entry/nav_plant_obs.spec.ts:111]: it is a good choice. I added a check which confirms successful submission.

**Pull request merged**